DependencyInjection/Configuration.php

```diff
@@ -116,6 +116,24 @@ public function getConfigTreeBuilder()
                     ->treatNullLike([])
                     ->defaultValue([])
                 ->end()
+                ->arrayNode('feature_flags')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('name')
+                                ->isRequired()
+                                ->validate()
+                                    ->ifTrue(function ($value) {
+                                        return !is_string($value);
+                                    })
+                                    ->thenInvalid('Feature flag name should be a string, got %s')
```

Comment:
Not in the Maze Runner tests — this check runs when Symfony compiles the configuration, which happens when the docker images are built, so it's too early for us to check with Maze Runner

We don't have any unit tests for this file, but I can look into how hard it'd be to add some

Reply:
Turns out it's very easy to unit test — I've added some for this case specifically as well as the default values and overriding values

I haven't added exhaustive testing for this because I think it'd be more useful to rework the `ClientFactoryTest` to load the configuration, so it's a bit more true to life

Reply:
oh neat! this is great.